tutorials/viewer.md

```diff
@@ -1,58 +1,296 @@
+
+Adding data with a dimensionality greater than 2D will cause dimension sliders to appear directly underneath the main canvas and above the status bar. As many sliders as needed will appear to ensure the data can be fully browsed. For example a 3D dataset needs one slider, a 4D dataset needs two sliders, and so on.
+
+It is also possible to mix data of different shapes and dimensionality in different layers. If a 2D and 4D dataset are both added to the viewer then the sliders will only affect the 4D dataset and the 2D dataset will be remain the same.
+
+For example the following commands from the console will add a 2D and 4D datasets to the same viewer:
+
+```python
+%gui qt5
+from skimage import data
+import napari
+
+viewer = napari.Viewer()
+viewer.add_image(moon, name='moon')
+blobs = np.stack(
+    [
+        data.binary_blobs(
+            length=128, blob_size_fraction=0.05, n_dim=3, volume_fraction=f
+        )
+        for f in np.linspace(0.05, 0.5, 10)
+    ],
+    axis=0,
+).astype(float)
+viewer.add_image(blobs, name='4D blobs')
+```
+
+![image](resources/multidimensional_sliders.gif)
+
+Notice how there are two dimensions sliders located beneath the main canvas and
+adjusting them affects only the `4D blobs` data.
+
+The multidimensional slides conclude our exploration of the layout of the viewer.
+
+## changing viewer theme
+
+Currently, **napari** comes with two different themes `light` and `dark`, which is the default.
+To change the theme just update `theme` property of the viewer, for example
 ```python
 viewer.theme = 'light'
 ```
 
-### Reordering layers on the viewer
+![image](resources/viewer_theme_light.png)
+
+Adding your own custom theme isn't too hard either but does require creating your own color `palette`
+and rebuilding the icons. If people want more themes, we're happy to add them or if you look checkout
+our [Contributing Guidelines](../CONTRIBUTING.md) for more information about building the icons and
+add one yourself!
+
+## custom keybinding
+
+One of the promises of **napari** is to provide a beginner friendly environment for interactive
+analysis. For example we want to enable workflows where people can interact with the gui, say
+click on the centers of some objects, or paint over some regions and then perform custom analysis.
+As a first step towards enabling custom interactivity we've provided support to add your own custom
+keybindings to the `Viewer` or individual `Layer` objects such that when the corresponding key gets
+clicked your custom function gets executed. Depending on which object you bind your key too your function
+will either get access to the state of the entire `viewer` or `layer` object.
+
+For example to bind function that loops through all layers in the viewer and prints their names to your console when you
+press the `p` key you can do the following:
```

Comment:
Is the problem fixed where keybindings only worked when the canvas was focused? If not, then this should be mentioned here I think.

Reply:
not yet, adding a mention